src/services/codefixes/fixAddMissingPropertyInitializer.ts

```diff
@@ -0,0 +1,112 @@
+        const checker = context.program.getTypeChecker();
+        const initializer = getInitializer(checker, propertyDeclaration);
+        if (!initializer) {
+            return undefined;
+        }
+
+        const newLineCharacter = getNewLineOrDefaultFromHost(context.host, context.formatContext.options);
+        const description = formatStringFromArgs(getLocaleSpecificMessage(Diagnostics.Declare_property_0), [token.text]);
+        const changes = textChanges.ChangeTracker.with(context, t => addInitializer(t, context.sourceFile, propertyDeclaration, initializer, newLineCharacter));
+        const action = { description, changes, fixId };
+        return [ action ];
+    }
+
+    function addInitializer (changeTracker: textChanges.ChangeTracker, propertyDeclarationSourceFile: SourceFile, propertyDeclaration: PropertyDeclaration, initializer: Expression, newLineCharacter: string): void {
+        const newNode = clone(propertyDeclaration);
+        newNode.initializer = initializer;
+        changeTracker.replaceNode(propertyDeclarationSourceFile, propertyDeclaration, newNode, { suffix: newLineCharacter });
+    }
+
+    function getInitializer(checker: TypeChecker, propertyDeclaration: PropertyDeclaration): Expression | undefined {
+        return getDefaultValueFromType(checker, checker.getTypeFromTypeNode(propertyDeclaration.type));
+    }
+
+    function getDefaultValueFromType (checker: TypeChecker, type: Type): Expression | undefined {
+        if (type.getFlags() & TypeFlags.String) {
+            return createLiteral("");
+        }
+        else if (type.getFlags() & TypeFlags.Number) {
+            return createNumericLiteral("0");
+        }
+        else if (type.getFlags() & TypeFlags.Boolean) {
+            return createFalse();
+        }
+        else if (type.getFlags() & TypeFlags.Literal) {
+            return createLiteral((<LiteralType>type).value);
+        }
+        else if (type.getFlags() & TypeFlags.Union) {
+            for (const t of (<UnionType>type).types) {
+                const initializer = getDefaultValueFromType(checker, t);
+                if (initializer) {
+                    return initializer;
+                }
+            }
+        }
+        else if (type.getFlags() & TypeFlags.Object) {
+            const objectType = <ObjectType>type;
+            if (getObjectFlags(objectType) & ObjectFlags.Class) {
+                const classDeclaration = <ClassLikeDeclaration>getClassLikeDeclarationOfSymbol(objectType.symbol);
+                const constructorDeclaration = find<ClassElement, ConstructorDeclaration>(classDeclaration.members, (m): m is ConstructorDeclaration => isConstructorDeclaration(m) && !!m.body)!;
+                if (constructorDeclaration &&
+                    (!constructorDeclaration.parameters || !constructorDeclaration.parameters.length) &&

```

Comment:
`parameters` should always be defined. Don't think you need to look at `typeParameters` since those are illegal on a constructor.